### PR TITLE
[6.4] UI tests for some host actions and create with parameters

### PR DESCRIPTION
**Pull request by ldjebran**
Depend on https://github.com/SatelliteQE/airgun/pull/179

```bash
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_host.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 9 items                                                                                           2018-08-15 09:41:43 - conftest - DEBUG - BZ deselect is disabled in settings

collected 9 items                                                                                            

tests/foreman/ui_airgun/test_host.py::test_positive_create PASSED                                      [ 11%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_details_page PASSED                      [ 22%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_edit_page PASSED                         [ 33%]
tests/foreman/ui_airgun/test_host.py::test_positive_delete PASSED                                      [ 44%]
tests/foreman/ui_airgun/test_host.py::test_positive_inherit_puppet_env_from_host_group_when_action PASSED [ 55%]
tests/foreman/ui_airgun/test_host.py::test_positive_create_host_with_parameters PASSED                 [ 66%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_organization PASSED                         [ 77%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_location PASSED                             [ 88%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_compliance_policy PASSED                    [100%]

========================================= 9 passed in 546.75 seconds =========================================
```

### CODE REVIEW

**Comment by ldjebran** [comment]
rebased 
```bash
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_host.py============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 9 items                                                                                           2018-08-15 10:52:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 9 items                                                                                            

tests/foreman/ui_airgun/test_host.py::test_positive_create PASSED                                      [ 11%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_details_page PASSED                      [ 22%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_edit_page PASSED                         [ 33%]
tests/foreman/ui_airgun/test_host.py::test_positive_delete PASSED                                      [ 44%]
tests/foreman/ui_airgun/test_host.py::test_positive_inherit_puppet_env_from_host_group_when_action PASSED [ 55%]
tests/foreman/ui_airgun/test_host.py::test_positive_create_host_with_parameters PASSED                 [ 66%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_organization PASSED                         [ 77%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_location PASSED                             [ 88%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_compliance_policy PASSED                    [100%]

========================================= 9 passed in 547.64 seconds =========================================

```

File: tests/foreman/ui_airgun/test_host.py [hunk 1]
```diff
@@ -223,3 +306,180 @@ def test_positive_inherit_puppet_env_from_host_group_when_action(session):
         values = session.host.read(host.name)
         assert values['host']['hostgroup'] == hostgroup.name
         assert values['host']['puppet_environment'] == env.name
+
+
+@tier3
```
Comment (rplevka) [inline tests/foreman/ui_airgun/test_host.py]: i'm not sure why this (and all the others) are marked as tier3 - the host is fake, i would go with t2

**Comment by ldjebran** [comment]
@rplevka changed and rebased

```bash
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_host.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 9 items                                                                                           2018-08-23 14:08:53 - conftest - DEBUG - BZ deselect is disabled in settings

collected 9 items                                                                                            

tests/foreman/ui_airgun/test_host.py::test_positive_create PASSED                                      [ 11%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_details_page PASSED                      [ 22%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_edit_page PASSED                         [ 33%]
tests/foreman/ui_airgun/test_host.py::test_positive_delete PASSED                                      [ 44%]
tests/foreman/ui_airgun/test_host.py::test_positive_inherit_puppet_env_from_host_group_when_action PASSED [ 55%]
tests/foreman/ui_airgun/test_host.py::test_positive_create_host_with_parameters PASSED                 [ 66%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_organization PASSED                         [ 77%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_location PASSED                             [ 88%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_compliance_policy PASSED                    [100%]

========================================= 9 passed in 780.52 seconds =========================================
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ 


```

**Comment by ldjebran** [comment]
@abalakh @rplevka tests output to latest changes in airgun
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_host.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 9 items                                                                                           2018-08-30 19:26:01 - conftest - DEBUG - BZ deselect is disabled in settings

collected 9 items                                                                                            

tests/foreman/ui_airgun/test_host.py::test_positive_create PASSED                                      [ 11%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_details_page PASSED                      [ 22%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_edit_page PASSED                         [ 33%]
tests/foreman/ui_airgun/test_host.py::test_positive_delete PASSED                                      [ 44%]
tests/foreman/ui_airgun/test_host.py::test_positive_inherit_puppet_env_from_host_group_when_action PASSED [ 55%]
tests/foreman/ui_airgun/test_host.py::test_positive_create_host_with_parameters PASSED                 [ 66%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_organization PASSED                         [ 77%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_location PASSED                             [ 88%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_compliance_policy PASSED                    [100%]

========================================= 9 passed in 781.54 seconds =========================================
```

**Comment by ldjebran** [comment]
test out put with last changes
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_host.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 9 items                                                                                           2018-09-03 17:15:56 - conftest - DEBUG - BZ deselect is disabled in settings

collected 9 items                                                                                            

tests/foreman/ui_airgun/test_host.py::test_positive_create PASSED                                      [ 11%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_details_page PASSED                      [ 22%]
tests/foreman/ui_airgun/test_host.py::test_positive_read_from_edit_page PASSED                         [ 33%]
tests/foreman/ui_airgun/test_host.py::test_positive_delete PASSED                                      [ 44%]
tests/foreman/ui_airgun/test_host.py::test_positive_inherit_puppet_env_from_host_group_when_action PASSED [ 55%]
tests/foreman/ui_airgun/test_host.py::test_positive_create_host_with_parameters PASSED                 [ 66%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_organization PASSED                         [ 77%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_location PASSED                             [ 88%]
tests/foreman/ui_airgun/test_host.py::test_positive_assign_compliance_policy PASSED                    [100%]

========================================= 9 passed in 900.00 seconds =========================================
```

**Comment by rplevka** [comment]
is it really worth spending 15 minutes on these 9 tests?
Don't take me wrong, the code looks good and everything,...but :)

let's take these two:
`tests/foreman/ui_airgun/test_host.py::test_positive_assign_organization`
`tests/foreman/ui_airgun/test_host.py::test_positive_assign_location`

does it really add so much value by treating these as individual tests? We create everything from a scratch for them and run separate new UI session for each.
Why not merge them together, call it let's say `test_positive_assign_taxonomies`
and do both assignments in a single go?
The test will fail if assigning loc assignment fails but org passes? - so what? Even if it fails, one still need to go in there and see what is going on, and it takes him like 2 seconds to figure out what works and what does not.

My point here is - why spend too much resources just to test the nitpicks?

**Comment by ldjebran** [comment]
@rplevka @lpramuk @abalakh joined that tests please revisit
```bash
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_host.py::test_positive_assign_taxonomies 
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collecting 1 item                                                                                            2018-09-04 14:56:08 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                             

tests/foreman/ui_airgun/test_host.py::test_positive_assign_taxonomies PASSED                           [100%]

========================================= 1 passed in 120.61 seconds =========================================
```

**Pull request merged**